src/App.js

```diff
@@ -1,14 +1,107 @@
-import React from 'react';
+import React, { Component } from "react";
+import TodoList from "./components/TodoList";
+import TodoForm from "./components/TodoForm";
+import MetaTags from "react-meta-tags";
+import styled from "styled-components";
+import { Container, Row } from "reactstrap";
+import "./App.css";
+
+const StyledDiv = styled.div`
+  border-top-left-radius: 42px;
+  background-color: white;
+  padding: 2em;
+`;
+const StyledH2 = styled.h2`
+font-family:  font-family: Arial, Helvetica, sans-serif;
+`;
+const StyledH6 = styled.h6`
+  color: gray;
+`;
+
+const LOCALSTORAGE_KEY = "todos";
+
+class App extends Component {
+  constructor() {
+    super();
+    const localStorageTodos = window.localStorage.getItem(LOCALSTORAGE_KEY);
+    this.state = {
+      todos: localStorageTodos ? JSON.parse(localStorageTodos) : [],
+    };
+  }
+
+  onWriteToLocalStorage = (todos) => {
+    const jsonTodos = JSON.stringify(todos);
+    window.localStorage.setItem(LOCALSTORAGE_KEY, jsonTodos);
+  };
+
+  onUpdateData = (todos) => {
+    this.setState({ todos });
+    this.onWriteToLocalStorage(todos);
+  };
+
+  addTodo = (val) => {
+    const newTodos = [
+      ...this.state.todos,
+      {
+        task: val,
+        id: Date.now(),
+        completed: false,
+      },
+    ];
+    console.log("Sunt in fuctia addTodo()");
+    this.onUpdateData(newTodos);
+  };
+
+  toggleCompleted = (id) => {
+    const updatedTodos = this.state.todos.map((todo) => {
+      if (todo.id === id) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+      return todo;
+    });
+    this.onUpdateData({ updatedTodos });
```

Comment:
okay, deci problema e ca atunci cand setezi in state o faci diferit pentru diferite metode.
Dupa aceasta metoda tu primesti in state:
```
todos: {
updatedTodos: [bla bla todos]
}
```

Reply:
pentru a o  elimina: `this.onUpdateData(updatedTodos);`